client/me/pending-payments/pending-list-item.jsx

```diff
@@ -3,33 +3,81 @@
+import { connect } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import Card from 'components/card';
 import Button from 'components/button';
+import { getSite, getSiteTitle, getSiteDomain } from 'state/sites/selectors';
+import PurchaseSiteHeader from '../purchases/purchases-site/header';
+import { purchaseType as getPurchaseType, getName } from 'lib/purchases';
+import { paymentMethodName } from 'lib/cart-values';
+import formatCurrency from 'lib/format-currency';
+
+export function PendingListItem( {
+	translate,
+	paymentType,
+	totalCost,
+	currency,
+	siteId,
+	siteTitle,
+	siteDomain,
+	dateCreated,
+	products,
+} ) {
+	let productName = '';
+	let purchaseType = '';
+
+	if ( products.length > 1 ) {
+		productName = 'Multiple Items';
+		purchaseType = 'Various';
+	} else if ( products[ 0 ] ) {
+		productName = getName( products[ 0 ] );
+		purchaseType = getPurchaseType( products[ 0 ] );
+	} else {
+		productName = '';
+		purchaseType = '';
+	}
 
-export function PendingListItem( { productName, paymentType, totalCostDisplay, translate } ) {
 	return (
-		<Card className={ 'pending-payments__list-item' }>
-			<span className="pending-payments__list-item-wrapper">
-				<div className="pending-payments__list-item-details">
-					<div className="pending-payments__list-item-title">{ productName }</div>
-					<div className="pending-payments__list-item-purchase-type">{ paymentType }</div>
-					<div className="pending-payments__list-item-purchase-cost">{ totalCostDisplay }</div>
-					<div className="pending-payments__list-item-actions">
-						<Button primary={ true } href="/help/contact">
-							<Gridicon icon="help" />
-							<span>{ translate( 'Contact Support' ) }</span>
-						</Button>
+		<React.Fragment>
+			<PurchaseSiteHeader siteId={ siteId } name={ siteTitle } domain={ siteDomain } />
+			<Card className={ 'pending-payments__list-item is-compact' }>
+				<span className="pending-payments__list-item-wrapper">
+					<div className="pending-payments__list-item-details">
+						<div className="pending-payments__list-item-title">{ productName }</div>
+						<div className="pending-payments__list-item-product">{ purchaseType }</div>
+						<div className="pending-payments__list-item-payment">
+							{ translate(
+								'Payment of %(totalCost)s was initiated on %(dateCreated)s via %(paymentType)s.',
+								{
+									args: {
+										totalCost: formatCurrency( totalCost, currency ),
+										dateCreated: moment( dateCreated ).format( 'LLL' ),
+										paymentType: paymentMethodName( paymentType ),
+									},
+								}
+							) }
+						</div>
+						<div className="pending-payments__list-item-actions">
+							<Button primary={ false } compact href="/help/contact">
+								<Gridicon icon="help" />
+								<span>{ translate( 'Contact Support' ) }</span>
+							</Button>
+						</div>
 					</div>
-				</div>
-			</span>
-		</Card>
+				</span>
+			</Card>
+		</React.Fragment>
 	);
 }
 
```

Comment:
It'd be nice to add proptypes. (I think it's basically the same as a stateful component - https://stackoverflow.com/a/44583006/1742789)